app/src/main/java/com/babylon/wallet/android/presentation/transaction/analysis/processor/ValidatorStakeProcessor.kt

```diff
@@ -27,77 +25,56 @@ class ValidatorStakeProcessor @Inject constructor(
-        executionSummary: ExecutionSummary,
-        assets: List<Asset>,
-        involvedOwnedAccounts: List<Account>
-    ) = executionSummary.deposits.map { depositsPerAccount ->
-        val defaultDepositGuarantees = getProfileUseCase().appPreferences.transaction.defaultDepositGuarantee
-        depositsPerAccount.value.map { depositedResource ->
-            val asset = assets.find {
-                it.resource.address == depositedResource.address
-            } ?: error("No asset found")
-            if (asset is LiquidStakeUnit) {
-                resolveLSU(asset, depositedResource, defaultDepositGuarantees)
-            } else {
-                executionSummary.resolveDepositingAsset(depositedResource, assets, defaultDepositGuarantees)
+    private fun List<AccountWithTransferables>.augmentWithStakes(
+        stakes: List<TrackedValidatorStake>
+    ): List<AccountWithTransferables> = map { accountWithTransferableResources ->
+        val transferables = accountWithTransferableResources.transferables.map tr@{ transferable ->
+            val lsu = (transferable as? Transferable.FungibleType.LSU) ?: return@tr transferable
+
+            var totalStakeForLSU = 0.toDecimal192()
+            var totalXrdWorthForLSU = 0.toDecimal192()
+
+            stakes.filter {
+                it.liquidStakeUnitAddress == lsu.resourceAddress
+            }.forEach { stake ->
+                totalStakeForLSU += stake.liquidStakeUnitAmount
+                totalXrdWorthForLSU += stake.xrdAmount
             }
-        }.toAccountWithTransferableResources(depositsPerAccount.key, involvedOwnedAccounts)
-    }
 
-    private fun DetailedManifestClass.ValidatorStake.resolveLSU(
-        asset: LiquidStakeUnit,
-        depositedResource: ResourceIndicator,
-        defaultDepositGuarantees: Decimal192
-    ): Transferable.Depositing {
-        val relatedStakes = validatorStakes.filter { it.liquidStakeUnitAddress == asset.resourceAddress }
-        val totalStakedLsuForAccount = relatedStakes.sumOf { it.liquidStakeUnitAmount }
-        val totalStakeXrdWorthForAccount = relatedStakes.sumOf { it.xrdAmount }
-        val lsuAmount = depositedResource.amount
-        val xrdWorth = ((lsuAmount / totalStakedLsuForAccount) * totalStakeXrdWorthForAccount).roundedWith(asset.resource.divisibility)
-        val guaranteeType = depositedResource.guaranteeType(defaultDepositGuarantees)
-        return Transferable.Depositing(
-            transferable = TransferableAsset.Fungible.LSUAsset(
-                amount = lsuAmount,
-                lsu = LiquidStakeUnit(asset.resource.copy(ownedAmount = lsuAmount), asset.validator),
-                xrdWorth = xrdWorth,
-            ),
-            guaranteeType = guaranteeType
-        )
+            val lsuAmount = when (lsu.amount) {
+                is FungibleAmount.Exact -> lsu.amount.amount
+                is FungibleAmount.Predicted -> lsu.amount.estimated
+                else -> TODO() // Cannot calculate
```

Comment:
Can you add please a comment explaining in short what we should do here?

Reply:
I mean that in a regular transaction you will only get an exact amount or an estimated amount.

Reply:
@sergiupuhalschi-rdx Pushed.